src/iop/sharpen.c

```diff
@@ -275,388 +277,144 @@ void tiling_callback(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t
 }
 
-#if defined(__SSE__)
-void process_sse2(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *const ivoid,
-                  void *const ovoid, const dt_iop_roi_t *const roi_in, const dt_iop_roi_t *const roi_out)
+void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *const ivoid,
+             void *const ovoid, const dt_iop_roi_t *const roi_in, const dt_iop_roi_t *const roi_out)
 {
+  if (!dt_iop_have_required_input_format(4 /*we need full-color pixels*/, self, piece->colors,
+                                         ivoid, ovoid, roi_in, roi_out))
+    return;
   const dt_iop_sharpen_data_t *const data = (dt_iop_sharpen_data_t *)piece->data;
-  const int ch = piece->colors;
   const int rad = MIN(MAXR, ceilf(data->radius * roi_in->scale / piece->iscale));
   // Special case handling: very small image with one or two dimensions below 2*rad+1 treat as no sharpening and just
   // pass through.  This avoids handling of all kinds of border cases below.
   if(rad == 0 ||
      (roi_out->width < 2 * rad + 1 || roi_out->height < 2 * rad + 1))
   {
-    dt_iop_image_copy_by_size(ovoid, ivoid, roi_out->width, roi_out->height, ch);
+    dt_iop_image_copy_by_size(ovoid, ivoid, roi_out->width, roi_out->height, 4);
     return;
   }
 
-  float *restrict tmp;
-  if (!dt_iop_alloc_image_buffers(self, roi_in, roi_out, 1, &tmp, 0))
+  float *restrict tmp;	// one row per thread
+  size_t padded_size;
+  if (!dt_iop_alloc_image_buffers(self, roi_in, roi_out,
+                                  1 | DT_IMGSZ_WIDTH | DT_IMGSZ_PERTHREAD, &tmp, &padded_size,
+                                  0))
   {
-    dt_iop_copy_image_roi(ovoid, ivoid, ch, roi_in, roi_out, TRUE);
+    dt_iop_copy_image_roi(ovoid, ivoid, 4, roi_in, roi_out, TRUE);
     return;
   }
 
   const int wd = 2 * rad + 1;
   const int wd4 = (wd & 3) ? (wd >> 2) + 1 : wd >> 2;
 
   const size_t mat_size = (size_t)4 * wd4;
-  float *const mat = dt_alloc_align_float(mat_size);
-  memset(mat, 0, sizeof(float) * mat_size);
-
   const float sigma2 = (1.0f / (2.5 * 2.5)) * (data->radius * roi_in->scale / piece->iscale)
                        * (data->radius * roi_in->scale / piece->iscale);
-  float weight = 0.0f;
-
-  // init gaussian kernel
-  for(int l = -rad; l <= rad; l++) weight += mat[l + rad] = expf(-l * l / (2.f * sigma2));
-  for(int l = -rad; l <= rad; l++) mat[l + rad] /= weight;
+  float *const mat = init_gaussian_kernel(rad, mat_size, sigma2);
 
-// gauss blur the image horizontally
+  const float *const restrict in = (float*)ivoid;
+  const size_t width = roi_out->width;
 #ifdef _OPENMP
 #pragma omp parallel for default(none) \
-  dt_omp_firstprivate(ch, ivoid, mat, rad, roi_in, roi_out, tmp, wd4) \
+  dt_omp_firstprivate(in, ovoid, mat, rad, width, roi_in, roi_out, tmp, padded_size) \
   schedule(static)
 #endif
+//  for(int j = rad; j < roi_out->height - rad; j++)
```

Comment:
probably to be removed, a leftovert from debug?

Reply:
Yup